src/napari_tiff/napari_tiff_metadata.py

```diff
@@ -259,45 +260,76 @@ def get_imagej_metadata(tif: TiffFile) -> dict[str, Any]:
     if res is not None:
         scale["Y"] = res.value[1] / max(res.value[0], 1)
     scale["Z"] = abs(ijmeta.get("spacing", 1.0))
-    if channel_axis is None:
-        scale = tuple(scale.get(x, 1.0) for x in axes if x != "S")
-    else:
-        scale = tuple(scale.get(x, 1.0) for x in axes if x not in "CS")
+    scale["T"] = ijmeta.get("finterval", 1.0)
+    unit_str = ijmeta.get('unit', 'pixel').encode().decode('unicode-escape')
+    scale_ = []
+    units = []
+    for ax in (x for x in axes if x not in 'CS'):
+        if ax == 'T':
+            scale_.append(scale.get('T'))
+            units.append('s')
+        else:
+            scale_.append(scale.get(ax, 1.0))
+            units.append(unit_str)
 
     kwargs = dict(
         rgb=rgb,
         channel_axis=channel_axis,
         name=name,
-        scale=scale,
+        scale=tuple(scale_),
         colormap=colormap,
         contrast_limits=contrast_limits,
         blending=blending,
         visible=visible,
+        units=tuple(units),
     )
     return kwargs
 
 
+def get_scale_and_units_from_ome(pixels: dict[str, Any], axes: str, shape: tuple[int, ...]) -> tuple[list[float], list[str]]:
+    pixel_size = []
+    units = []
+
+    for i, ax in enumerate(axes):
+        if ax == "c":
+            continue
+        if ax == 't':
+            if "TimeIncrement" not in pixels or "TimeIncrementUnit" not in pixels:
+                if shape[i] > 1:
+                    return [], []
```

Comment:
I don't understand why this is like this? (same below)

It might be worth adding a docstring to this function that explains the contents of "pixels", in which cases the unit returned is the default 'pixel' and in which cases the return is empty.

Reply:
Changed a little. Now it reflects the original code. 
Or you want to put 1 when Increment is missed?

Reply:
Yeah I would include it as "1 pixel" when it's missing. This prevents silent dropping of potentially useful metadata because *some* metadata is missing. Using 1 pixel would let you recover what metadata is there while including clear markers for users about what metadata is missing. See my comment below on the test parameters for what I would expect to happen.